### PR TITLE
Enable TTLAfterFinished e2e test

### PR DESCRIPTION
**Pull request by janetkuo**
**What this PR does / why we need it**:
This test is only running in test job `ci-kubernetes-e2e-gci-gce-alpha-features`, with all alpha APIs enabled:
https://github.com/kubernetes/test-infra/blob/0053c6db9437e1ad888cbf2e4a1a17e9425d5644/config/jobs/kubernetes/sig-gcp/sig-gcp-gce-config.yaml#L193

Therefore, remove the feature gate check to avoid the need to pass `--feature-gates=TTLAfterFinished=true` to test_args. 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/common/ttlafterfinished.go [hunk 1]
```diff
@@ -36,10 +35,8 @@ const dummyFinalizer = "k8s.io/dummy-finalizer"
 var _ = framework.KubeDescribe("TTLAfterFinished", func() {
 	f := framework.NewDefaultFramework("ttlafterfinished")
 
-	It("Job should be deleted once it finishes after TTL seconds [Feature:TTLAfterFinished]", func() {
-		if !utilfeature.DefaultFeatureGate.Enabled(features.TTLAfterFinished) {
-			framework.Skipf("Skip because %s feature is not enabled; run test with --feature-gates=%s=true", features.TTLAfterFinished, features.TTLAfterFinished)
-		}
+	alphaFeatureStr := "[Feature:TTLAfterFinished]"
+	It(fmt.Sprintf("Job should be deleted once it finishes after TTL seconds", alphaFeatureStr), func() {
```
Comment (enisoc) [inline test/e2e/common/ttlafterfinished.go]: The format string needs a `%s` at the end. Also just curious, what's the reason for breaking it out into a separate var?

Comment (janetkuo) [reply test/e2e/common/ttlafterfinished.go]: Fixed. Breaking it out just to make it more obvious that this is tested in the alpha feature suite.

**Comment by janetkuo** [comment]
Of course I forgot to update bazel! Reapplying tag.

**Pull request merged**